mlir/test/Integration/Dialect/SparseTensor/CPU/block_majors.mlir

```diff
@@ -0,0 +1,174 @@
+//--------------------------------------------------------------------------------------------------
+// WHEN CREATING A NEW TEST, PLEASE JUST COPY & PASTE WITHOUT EDITS.
+//
+// Set-up that's shared across all tests in this directory. In principle, this
+// config could be moved to lit.local.cfg. However, there are downstream users that
+// do not use these LIT config files. Hence why this is kept inline.
+//
+// DEFINE: %{sparsifier_opts} = enable-runtime-library=true
+// DEFINE: %{sparsifier_opts_sve} = enable-arm-sve=true %{sparsifier_opts}
+// DEFINE: %{compile} = mlir-opt %s --sparsifier="%{sparsifier_opts}"
+// DEFINE: %{compile_sve} = mlir-opt %s --sparsifier="%{sparsifier_opts_sve}"
+// DEFINE: %{run_libs} = -shared-libs=%mlir_c_runner_utils,%mlir_runner_utils
+// DEFINE: %{run_opts} = -e main -entry-point-result=void
+// DEFINE: %{run} = mlir-cpu-runner %{run_opts} %{run_libs}
+// DEFINE: %{run_sve} = %mcr_aarch64_cmd --march=aarch64 --mattr="+sve" %{run_opts} %{run_libs}
+//
+// DEFINE: %{env} =
+//--------------------------------------------------------------------------------------------------
+
+// RUN: %{compile} | %{run} | FileCheck %s
+//
+// Do the same run, but now with direct IR generation.
+// REDEFINE: %{sparsifier_opts} = enable-runtime-library=false
+// RUN: %{compile} | %{run} | FileCheck %s
+//
+// Do the same run, but now with direct IR generation and vectorization.
+// REDEFINE: %{sparsifier_opts} = enable-runtime-library=false enable-buffer-initialization=true vl=2 reassociate-fp-reductions=true enable-index-optimizations=true
+// RUN: %{compile} | %{run} | FileCheck %s
+
+#BSR_row_rowmajor = #sparse_tensor.encoding<{
+  map = (i, j) ->
+    ( i floordiv 3 : dense
+    , j floordiv 4 : compressed
+    , i mod 3 : dense
+    , j mod 4 : dense
+    )
+}>
+
+#BSR_row_colmajor = #sparse_tensor.encoding<{
+  map = (i, j) ->
+    ( i floordiv 3 : dense
+    , j floordiv 4 : compressed
+    , j mod 4 : dense
+    , i mod 3 : dense
+    )
+}>
+
+#BSR_col_rowmajor = #sparse_tensor.encoding<{
+  map = (i, j) ->
+    ( j floordiv 4 : dense
+    , i floordiv 3 : compressed
+    , i mod 3 : dense
+    , j mod 4 : dense
+    )
+}>
+
+#BSR_col_colmajor = #sparse_tensor.encoding<{
+  map = (i, j) ->
+    ( j floordiv 4 : dense
+    , i floordiv 3 : compressed
+    , j mod 4 : dense
+    , i mod 3 : dense
+    )
+}>
+
+//
+// Example 3x4 block storage of a 6x16 matrix:
+//
+//  +---------+---------+---------+---------+
+//  | 1 2 . . | . . . . | . . . . | . . . . |
+//  | . . . . | . . . . | . . . . | . . . . |
+//  | . . . 3 | . . . . | . . . . | . . . . |
+//  +---------+---------+---------+---------+
+//  | . . . . | . . . . | 4 5 . . | . . . . |
+//  | . . . . | . . . . | . . . . | . . . . |
+//  | . . . . | . . . . | . . 6 7 | . . . . |
+//  +---------+---------+---------+---------+
+//
+// Stored for row "2x4-blocked"
```

Comment:
Is it 3x4 and 4x3 instead of 2x4 and 4x2? And can we say row-major and column-major? It took me a second to understand.
And can you explain a bit about why there is `1` in positions[1] and coordinates[1] instead of just positions array and coordinates array?

Reply:
No it is really 2x4 and 4x2. Look at "blocks" as single elements. Then you get a 2x4 and 4x2 CSR or CSC storage of the blocks. Within the blocks we have a 3x4 or 4x3 storage, giving a 6x16 overall. But I will update the comment a bit more

Reply:
Ah I see! That makes sense. Thanks for the explanation.